ProofWidgets/Component/FilterDetails.lean

```diff
@@ -0,0 +1,21 @@
+import ProofWidgets.Data.Html
+
+open Lean ProofWidgets
+
+/-- The props for the `FilterDetails` component. -/
+structure FilterDetailsProps where
+  /-- The title -/
+  message : String
+  /-- What is shown in the filtered state -/
+  filtered : Html
```

Comment:
In your use case @JovanGerb , are the `filtered` and `all` fields quick to compute? If not (and even if yes, but there were other uses where they are not quick), it might make sense to make this a lazy `Bool -> Html` or `Bool -> IO Html` computation so that the unfiltered version is not computed if the user doesn't click the filter button.

Reply:
In my case they aren't quick to compute, but the vast majority of computation time is spent on a computation that is needed for both the fields. So I think it makes more sense to supply both fields immediately. Otherwise this intermediate result would have to be stored as a state.

Reply:
Hmm, if it was a function and you already had the two values computed, you could use `(fun b => if b then filtered else all)`. However, it would be a bit of work to get thunk output caching to work properly so let's not optimize prematurely.